### PR TITLE
Kate fixes

### PR DESCRIPTION
**Pull request by hknaack**
Reduce restrictions, so that Kate can be used as intended by the developers:

- allow usage of Git
- permit Bash completion in built-in Konsole
- permit the requested access to kwinrc configuration

### CODE REVIEW

File: etc/profile-a-l/kate.profile
Comment (rusty-snake) [inline etc/profile-a-l/kate.profile:9]: `noblacklist`

File: etc/profile-a-l/kate.profile
Comment (rusty-snake) [inline etc/profile-a-l/kate.profile:28]:
```suggestion
include allow-common-devel.inc

```

**Comment by hknaack** [comment]
Kate remembers where it was previously closed. When starting Kate at a later time, it will automatically restore that previous session. If that location contains a version control system like Git, Kate attempts to read the repository information.
At the moment, this process seems to be blocked by firejail. The Projects-sidebar stays blank during the first 30 seconds after start of Kate. After that, the directory structure appears and console output reports: `QProcess: Destroyed while process ("/usr/bin/git") is still running.`
So far, it just seems to be just an annoying delay (which prevents from accessing one of the displayed files in the sidebar).
Any suggestion appreciated.

**Comment by hknaack** [comment]
I haven't investigated any further, what is blocking git at start-up and causing the initial delay. But it is mainly annoying and not a show-stopper. So I mark this request as ready for review and maybe address the git issue in a future pull request.

**Review by rusty-snake — APPROVED** [review]

**Review by glitsj16 — APPROVED** [review]
LGTM

**Comment by glitsj16** [comment]
> I haven't investigated any further, what is blocking git at start-up and causing the initial delay. But it is mainly annoying and not a show-stopper. So I mark this request as ready for review and maybe address the git issue in a future pull request.

Here's something you might be interested in when looking into adding git support for Kate later on. I didn't fully test this, basically compared kate and git profiles and added what seems relevant in this context. You can put it in your kate.local for testing.

```
## Allow git
#+ blacklisted by disable-common.inc
noblacklist ${HOME}/.git-credential-cache
noblacklist ${HOME}/.git-credentials
ignore rmenv GH_TOKEN
ignore rmenv GITHUB_TOKEN
ignore rmenv GH_ENTERPRISE_TOKEN
ignore rmenv GITHUB_ENTERPRISE_TOKEN
#+ blacklisted by disable-programs.inc
noblacklist ${HOME}/.config/git
noblacklist ${HOME}/.gitconfig
noblacklist ${HOME}/.gnupg

# Allow ssh (blacklisted by disable-common.inc)
include allow-ssh.inc

# Allow networking
protocol unix,inet,inet6
ignore protocol

```

**Comment by hknaack** [comment]
> > I haven't investigated any further, what is blocking git at start-up and causing the initial delay. But it is mainly annoying and not a show-stopper. So I mark this request as ready for review and maybe address the git issue in a future pull request.
> 
> Here's something you might be interested in when looking into adding git support for Kate later on. I didn't fully test this, basically compared kate and git profiles and added what seems relevant in this context. You can put it in your kate.local for testing.
> 
> ```
> ## Allow git
> #+ blacklisted by disable-common.inc
> noblacklist ${HOME}/.git-credential-cache
> noblacklist ${HOME}/.git-credentials
> ignore rmenv GH_TOKEN
> ignore rmenv GITHUB_TOKEN
> ignore rmenv GH_ENTERPRISE_TOKEN
> ignore rmenv GITHUB_ENTERPRISE_TOKEN
> #+ blacklisted by disable-programs.inc
> noblacklist ${HOME}/.config/git
> noblacklist ${HOME}/.gitconfig
> noblacklist ${HOME}/.gnupg
> 
> # Allow ssh (blacklisted by disable-common.inc)
> include allow-ssh.inc
> 
> # Allow networking
> protocol unix,inet,inet6
> ignore protocol
> ```

Tried it, but it did not change the delay.

**Comment by glitsj16** [comment]
> Tried it, but it did not change the delay.

Ah, I think `tracelog` is responsible for the delay. When I take that out by adding `ignore tracelog` to what I suggested above the error `QProcess: Destroyed while process ("/usr/bin/git") is still running.` goes away.

Side-note: I noticed kate supports git-cola. You could add `
noblacklist ${HOME}/.config/git-cola` too IMO.

**Comment by hknaack** [comment]
I can confirm that the delay disappeared after disabling `tracelog` just on top of the commits in this PR.

From what I understand, this change would prevent us from getting noticed about attempts to access blocked files/directorys. From what I've seen, this is quite often used in profiles. Is it safe to remove then, or why is it required beyond debugging of profiles?

Regarding git-cola: I haven't installed or used it yet. However, Kate also supports tools like Qgit, Giggle, gitg and gitk. So, I would prefer to leave this to someone who has got experience using git-cola.

**Comment by rusty-snake** [comment]
> Is it safe to remove then, or why is it required beyond debugging of profiles?

If it causes trouble remove it, otherwise add it.

**Comment by kmk3** [comment]
@hknaack FYI, the author name/email and committer name/email are different on
the commits (see with `git show --pretty=fuller`).  If this is not intentional,
you can reset them by e.g.: rebasing with
`git rebase -i "$(git merge-base master HEAD)"` and then using
`git commit --amend --reset-author` on each commit.  This helps ensure that
GitHub attributes the commits to your user.

**Comment by kmk3** [comment]
@hknaack The author/committer emails are the same now, but while the previous
"noreply" email was being linked to your user, the normal email from the
current commits isn't.  See the lack of linking to the user on:
https://github.com/netblue30/firejail/pull/5159/commits

Have you [added/verified][1] the normal email on GitHub?

[1]: https://github.com/settings/emails

**Comment by hknaack** [comment]
> Have you [added/verified](https://github.com/settings/emails) the normal email on GitHub?

Appreciated, but I prefer not to do that.

**Review by kmk3 — APPROVED** [review]

**Comment by kmk3** [comment]
@hknaack commented [on May 29](https://github.com/netblue30/firejail/pull/5159#issuecomment-1140414079):

> > Have you [added/verified](https://github.com/settings/emails) the normal
> > email on GitHub?
> 
> Appreciated, but I prefer not to do that.

Fair enough.  Just note that commits done from the web UI (such as file edits
and merges) may have a different email compared to the commits that are made
locally (unless both the local and GitHub emails are set to e.g.: the "noreply"
GitHub email).  Also, it is possible to verify an email without making it
public on the GitHub profile.

**Comment by netblue30** [comment]
thanks for the patch!

**Pull request merged**
